dsa-solutions/lc-solutions/0200-0299/0244-shortest-word-distance-II.md

```diff
@@ -0,0 +1,197 @@
+        
+        int i = 0, j = 0;
+        int min_distance = INT_MAX;
+        
+        while (i < positions1.size() && j < positions2.size()) {
+            min_distance = min(min_distance, abs(positions1[i] - positions2[j]));
+            if (positions1[i] < positions2[j]) {
+                ++i;
+            } else {
+                ++j;
+            }
+        }
+        
+        return min_distance;
+    }
+};
+```
+
+### Java
+```java
+class WordDistance {
+    private Map<String, List<Integer>> wordPositions;
+    
+    public WordDistance(String[] wordsDict) {
+        wordPositions = new HashMap<>();
+        for (int i = 0; i < wordsDict.length; i++) {
+            if (!wordPositions.containsKey(wordsDict[i])) {
+                wordPositions.put(wordsDict[i], new ArrayList<>());
+            }
+            wordPositions.get(wordsDict[i]).add(i);
+        }
+    }
+    
+    public int shortest(String word1, String word2) {
+        List<Integer> positions1 = wordPositions.get(word1);
+        List<Integer> positions2 = wordPositions.get(word2);
+        
+        int i = 0, j = 0;
+        int minDistance = Integer.MAX_VALUE;
+        
+        while (i < positions1.size() && j < positions2.size()) {
+            int pos1 = positions1.get(i);
+            int pos2 = positions2.get(j);
+            
+            minDistance = Math.min(minDistance, Math.abs(pos1 - pos2));
+            
+            if (pos1 < pos2) {
+                i++;
+            } else {
+                j++;
+            }
+        }
+        
+        return minDistance;
+    }
+}
+```
+
+### C
+```c
+typedef struct {
+    int* positions;
+    int count;
+} WordInfo;
+
+typedef struct {
+    WordInfo* wordMap;
+    int size;
+} WordDistance;
+
+WordDistance* wordDistanceCreate(char ** wordsDict, int wordsDictSize) {
+    WordDistance* obj = (WordDistance*)malloc(sizeof(WordDistance));
+    obj->wordMap = (WordInfo*)calloc(wordsDictSize, sizeof(WordInfo));
+    obj->size = wordsDictSize;
+    
+    for (int i = 0; i < wordsDictSize; i++) {
+        char* word = wordsDict[i];
+        WordInfo* wordInfo = &obj->wordMap[i];
+        wordInfo->count = 0;
+        wordInfo->positions = (int*)malloc(wordsDictSize * sizeof(int));
+        for (int j = 0; j < wordsDictSize; j++) {
+            if (strcmp(word, wordsDict[j]) == 0) {
+                wordInfo->positions[wordInfo->count++] = j;
+            }
+        }
+    }
+    
+    return obj;
+}
+
+int wordDistanceShortest(WordDistance* obj, char * word1, char * word2) {
+    WordInfo* word1Info = NULL;
+    WordInfo* word2Info = NULL;
+    
+    for (int i = 0; i < obj->size; i++) {
+        if (strcmp(word1, word2) == 0 && strcmp(word1, word2) == 0) {
+            word1Info = &obj->wordMap[i];
+            word2Info = word1Info;
+            break;
+        }
+        if (strcmp(word1, obj->wordMap[i].name) == 0) {
+             Returns Solution  I O also have no   even ActionController know
```

Comment:
Please ensure that you complete your code, and I strongly recommend that you follow our structure at https://github.com/orgs/CodeHarborHub/discussions/3369.